app/components/slide_up_panel/slide_up_panel.js

```diff
@@ -47,189 +47,241 @@ export default class SlideUpPanel extends PureComponent {
-        const {endPosition, finalPosition} = this.state;
-
-        if (finalPosition !== endPosition || forceCheck || (isGoingDown && this.canDrag)) {
-            const moveTravelledFarEnough = Math.abs(distanceY) > Math.abs(distanceX) && Math.abs(distanceY) > 2;
-            return moveTravelledFarEnough;
-        }
-
-        return false;
-    };
-
-    moveStart = (gestureState) => {
-        if (this.viewRef && this.backdrop) {
-            const {endPosition} = this.state;
-            const position = endPosition - (gestureState.y0 - gestureState.moveY);
-            this.isDragging = true;
-
-            this.backdrop.setNativeProps({pointerEvents: 'none'});
-            this.updatePosition(position);
-        }
+    closeWithAnimation = () => {
+        Animated.timing(this.translateYOffset, {
+            duration: 200,
+            toValue: this.snapPoints[2],
+            useNativeDriver: true,
+        }).start(() => this.props.onRequestClose());
     };
 
-    moveFinished = (gestureState) => {
-        if (this.viewRef) {
-            const isGoingDown = gestureState.y0 < gestureState.moveY;
-            let position = gestureState.moveY;
-            if (this.previousTop !== position) {
-                position = this.previousTop;
-            }
-
-            this.startAnimation(gestureState.y0, position, isGoingDown);
+    onHeaderHandlerStateChange = ({nativeEvent}) => {
+        if (nativeEvent.oldState === GestureState.BEGAN) {
+            this.lastScrollY.setValue(0);
         }
+        this.onHandlerStateChange({nativeEvent});
     };
 
-    setBackdropRef = (ref) => {
-        this.backdrop = ref;
-    };
-
-    setDrag = (val) => {
-        this.canDrag = val;
-    };
-
-    setViewRef = (ref) => {
-        this.viewRef = ref;
-    };
-
-    startAnimation = (initialY, positionY, isGoingDown, initial = false) => {
-        const {allowStayMiddle, containerHeight, onRequestClose} = this.props;
-        const {finalPosition, initialPosition} = this.state;
-        const position = new Animated.Value(initial ? initialY : positionY);
-        let endPosition = (!isGoingDown && !initial ? finalPosition : positionY);
-
-        position.removeAllListeners();
-        if (isGoingDown) {
-            if (positionY <= this.state.initialPosition && allowStayMiddle) {
-                endPosition = initialPosition;
+    onHandlerStateChange = ({nativeEvent}) => {
+        if (nativeEvent.oldState === GestureState.ACTIVE) {
+            const {translationY, velocityY} = nativeEvent;
+            const {allowStayMiddle, containerHeight} = this.props;
+            const {lastSnap} = this.state;
+            const isGoingDown = translationY > 0;
+            const translation = translationY - this.lastScrollYValue;
+
+            const endOffsetY = lastSnap + translation;
+            let destSnapPoint = this.snapPoints[0];
+
+            if (Math.abs(translationY) < 50) {
+                destSnapPoint = lastSnap;
```

Comment:
I want to make sure that I understand the logic here. This makes it so it doesn't move the panel until you've dragged 50 pixels

Reply:
yes in case it moves less than 50 points we snap it to the last position it was snapped before